federation/source-config-handlers/src/main/java/org/codice/ddf/admin/sources/impl/test/ValidUrlTestMethod.java

```diff
@@ -33,27 +37,33 @@
 public class ValidUrlTestMethod extends TestMethod<SourceConfiguration> {
 
     private final RequestUtils requestUtils = new RequestUtils();
-
-    private static final String DESCRIPTION = "Attempts to connect to a given hostname and port";
-
+    private static final String DESCRIPTION = "Attempts to connect to a given hostname and port or url";
     public static final String VALID_URL_TEST_ID = "valid-url";
 
-    private static final List<String> REQUIRED_FIELDS = ImmutableList.of(SOURCE_HOSTNAME, PORT);
+    private static final List<String> OPTIONAL_FIELDS = ImmutableList.of(SOURCE_HOSTNAME, PORT, ENDPOINT_URL);
     private static final Map<String, String> SUCCESS_TYPES = getCommonSourceSubtypeDescriptions(CONNECTED);
     private static final Map<String, String> FAILURE_TYPES = getCommonSourceSubtypeDescriptions(CANNOT_CONNECT);
 
     public ValidUrlTestMethod() {
         super(VALID_URL_TEST_ID,
                 DESCRIPTION,
-                REQUIRED_FIELDS,
                 null,
+                OPTIONAL_FIELDS,
                 SUCCESS_TYPES,
                 FAILURE_TYPES,
                 null);
     }
 
     @Override
     public Report test(SourceConfiguration configuration) {
-        return requestUtils.endpointIsReachable(configuration.sourceHostName(), configuration.sourcePort());
+        return configuration.endpointUrl() != null ?
+                requestUtils.endpointIsReachable(configuration.endpointUrl()) :
+                requestUtils.endpointIsReachable(configuration.sourceHostName(), configuration.sourcePort());
+    }
+    @Override
+    public List<ConfigurationMessage> validateOptionalFields(SourceConfiguration configuration){
```

Comment:
Formatting is off on this line. There are a few other minor formatting issues in the PR.